### PR TITLE
EZP-24745: Make SolrBundle ready for release

### PR DESCRIPTION
**Pull request by andrerom**
https://jira.ez.no/browse/EZP-24745

review ping @pspanja @bdunogier @lolautruche 

Todo:
- [x] update installer ezpublish.yml config
 - [x] https://github.com/ezsystems/ezpublish-kernel/pull/1389
 - [x] https://github.com/ezsystems/DemoBundle/pull/177
- Bugs encountered:
 - [x] Console command default siteaccess https://github.com/ezsystems/ezpublish-kernel/pull/1395
 - [x] Instance of almighty content type with invalid data on it and without location 
- [x] Docker setup https://github.com/ezsystems/ezpublish-docker/pull/40
- [x] DOC: INSTALL.md and https://doc.ez.no/display/EZP/Search+Engines

### CODE REVIEW

File: ezpublish/EzPublishKernel.php [hunk 1]
```diff
@@ -49,6 +49,7 @@ public function registerBundles()
             new EzSystems\PlatformInstallerBundle\EzSystemsPlatformInstallerBundle(),
             new EzSystems\PrivacyCookieBundle\EzSystemsPrivacyCookieBundle(),
             new EzSystems\RepositoryFormsBundle\EzSystemsRepositoryFormsBundle(),
+            new eZ\Bundle\EzPublishSolrSearchEngineBundle\EzPublishSolrSearchEngineBundle()
```
Comment (andrerom) [inline ezpublish/EzPublishKernel.php]: Name should probably be `EzSystems\PlatformSolrSearchEngineBundle\EzSystemsPlatformSolrSearchEngineBundle(),`, or?

**Comment by bdunogier** [comment]
The approach looks okay.

File: ezpublish/config/config.yml [hunk 2]
```diff
@@ -14,6 +15,36 @@ doctrine:
...
+            core: collection1
+    connections:
+        default:
+            entry_endpoints:
+                - endpoint0
+            mapping:
+                default: endpoint0
+
+# Stash is used for persistence cache in eZ Repository
+# Can have several caches used by different eZ Reposiotries in ezpublish.yml, must be one per database
```
Comment (lolautruche) [inline ezpublish/config/config.yml]: `Repositories`

Comment (andrerom) [inline ezpublish/config/config.yml]: ops, then I have some commits to update :P

**Comment by pspanja** [comment]
Looks good so far.

**Comment by andrerom** [comment]
@bdunogier How do we deal with the binary file exception issue in master? Currently anyone trying solr is forced to comment out https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/IO/TolerantIOService.php#L77

**Comment by bdunogier** [comment]
Do we know why this is happening with Solr, and not with other parts of the
system ?

**Comment by andrerom** [comment]
BD: no idea, first step is to fix demo data, second is to consider having script for upgrades before stable release.. :/

**Comment by andrerom** [comment]
"Simple", the given content does not have a location. @clash82 Can we remove this\* instance of Almighty from the demo data?

\* *content id 126, it does not have any locations so it's not causing issues in UI, but it causes issues when trying to index database with solr, see link to DemoBundle PR above*

**Comment by clash82** [comment]
> Can we remove this* instance of Almighty from the demo data

I think this can be removed. It shouldn't been there. I'll check it and create an update.

**Comment by clash82** [comment]
Extra feature ;) removed in https://github.com/ezsystems/DemoBundle/pull/179

**Comment by andrerom** [comment]
Ready for review now, ping @bdunogier @glye @lolautruche @michalpipa

File: INSTALL.md [hunk 3]
```diff
@@ -17,13 +17,22 @@
 
 ## Install
 
-0. **Create a clean database**
+0. **Create a empty database**, and optionally setup Solr
```
Comment (glye) [inline INSTALL.md]: Not sure if relevant for this PR, but same as for the other INSTALL.md I commented on this one also has a mismatch between header number in the source and when rendered, and a "skip to step 3" part which should say "skip to step 4".

**Comment by glye** [comment]
+1 AFAICT, apart from INSTALL.md note.

File: INSTALL.md (hunk 3 above)
Comment (andrerom) [inline INSTALL.md]: I fixed that in the other PR, can't fix it in both :)

**Comment by bdunogier** [comment]
+1 besides maybe the small text suggestions.

**Comment by andrerom** [comment]
Fixed the suggested text addition, thanks!

**Pull request merged**